cli/tests/integration_tests.rs

```diff
@@ -326,10 +326,86 @@ fn bundle_single_module() {
   assert_eq!(output.stderr, b"");
 }
 
-// TODO(#2933): Rewrite this test in rust.
 #[test]
 fn repl_test() {
-  util::run_python_script("tools/repl_test.py")
+  use std::io::Write;
+  use std::process::Stdio;
+  let repl_process = |lines: &Vec<&str>,
+                      envs: Option<Vec<(String, String)>>,
+                      stdout_str: &str,
+                      stderr_str: &str,
+                      exit_code: i32| {
+    let mut deno_process_builder = util::deno_cmd();
+    deno_process_builder
+      .stdin(Stdio::piped())
+      .stdout(Stdio::piped())
+      .stderr(Stdio::piped());
+    if let Some(envs) = envs {
+      deno_process_builder.envs(envs);
+    }
+    let mut deno = deno_process_builder
+      .spawn()
+      .expect("failed to spawn script");
+    {
+      let stdin = deno.stdin.as_mut().expect("failed to get stdin");
+      stdin
+        .write_all(lines.join("\n").as_bytes())
+        .expect("failed to write to stdin");
+    }
+    let output = deno.wait_with_output().expect("failed to wait on child");
+    assert!(util::pattern_match(
+      stdout_str,
+      std::str::from_utf8(&output.stdout).unwrap(),
+      "[WILDCARD]"
+    ));
+    assert!(util::pattern_match(
+      stderr_str,
+      std::str::from_utf8(&output.stderr).unwrap(),
+      "[WILDCARD]"
+    ));
+    assert_eq!(output.status.code().unwrap(), exit_code);
+  };
+  let test_cases = vec![
+		(
+			vec!["console.log('hello')", "'world'"],
+			None,
+			"hello\nundefined\nworld\n",
+			"",
+			0,
+		),
+		(vec!["1 + 2"], None, "3\n", "", 0),
+		(vec!["exit", "'ignored'"], None, "", "", 0),
+		(
+			vec!["help"],
+			None,
+			"_       Get last evaluation result\n_error  Get last thrown error\nexit    Exit the REPL\nhelp    Print this help message\n",
+			"",
+			0,
+		),
+		(vec!["Deno.writeFileSync"], None, "[Function: writeFileSync]\n", "", 0),
+		(vec!["(\n1 + 2\n)"], None, "3\n", "", 0),
+		(vec!["eval('{')"], None, "", "[WILDCARD]Unexpected end of input[WILDCARD]", 0),
+		(vec!["not_a_variable"], None, "", "[WILDCARD]not_a_variable is not defined[WILDCARD]", 0),
+		(vec!["syntax error"], None, "", "[WILDCARD]Unexpected identifier[WILDCARD]", 0),
+		(vec!["console()"], None, "", "[WILDCARD]console is not a function[WILDCARD]", 0),
+		(vec!["console()"], None, "", "[WILDCARD]console is not a function[WILDCARD]", 0),
+		(vec!["var a = 123;", "a"], None, "undefined\n123\n", "", 0),
+		(vec!["let a = 123;", "a"], None, "undefined\n123\n", "", 0),
+		(vec!["1"], Some(vec![("DENO_DIR".to_owned(), "nonexistent".to_owned())]), "1\n", "", 0),
+		(vec!["1", "_"], None, "1\n1\n", "", 0),
+		(vec!["throw 1", "_error"], None, "1\n", "Thrown: 1\n", 0),
+		(vec!["_ = 1", "2", "_"], None, "Last evaluation result is no longer saved to _.\n1\n2\n1\n", "", 0),
+		(vec!["_error = 1", "throw 2", "_error"], None, "Last thrown error is no longer saved to _error.\n1\n1\n", "Thrown: 2\n", 0),
+	];
```

Comment:
Can this be broken into individual test cases? Maybe there's a problem with them running in parallel tho.

Reply:
I thought the same while porting the tests but wasn't sure how to proceed so I tried to keep the port as close to the original .py file as possible. 
I don't think that it would be a problem running them in parallel.
What do you think of moving each test to an individual one namining them the same as the original python name and prefixing each one with `repl_`. It would also be necessary to add helping utils to the util mod.

Example:
```
def test_help_command(self):
    // ...
```
Would become 
```
#[test]
fn repl_test_help_command() {
    // ...
```

Reply:
@jsouto18 Yes I think that would be ideal. Just lift the `repl_process()` to the top-level.

Reply:
If there are problems with running them in parallel, then we could use a mutex lock of some sort... But ideally we don't have to have that complexity.